Intersect.Client/Entities/Resource.cs

```diff
@@ -15,14 +15,14 @@
 namespace Intersect.Client.Entities
 {
 
-    public partial class Resource : Entity
+    public partial class Resource : Entity, IResource
     {
 
         private bool _waitingForTilesets;
 
-        public ResourceBase BaseResource;
+        public ResourceBase BaseResource { get; set; }
 
-        public bool IsDead;
+        public bool IsDead { get; set; }
```

Comment:
Should this be exposed by the interface?

Reply:
I figured it could be?

Was thinking someone might want to figure out if a resource is consumed to render something on top of it (ie the tree falling over) or do something else with said information. Maybe render a shine over it when it's not consumed but stop rendering that when it's consumed?

Just a few random use cases.